src/core/varinfo.jl

```diff
@@ -206,6 +206,18 @@ push!(vi::VarInfo, vn::VarName, r::Any, dist::Distributions.Distribution, gid::I
   vi
 end
 
+addorder!(vi::VarInfo, vn::VarName, num_produce::Int) = begin
+  push!(vi.orders, num_produce)
+  vi
+end
+
+setorder!(vi::VarInfo, vn::VarName, index::Int) = begin
+  if vi.orders[vi.idcs[vn]] != index
```

Comment:
Why do you need to check this condition? Couldn't you just overwrite it

Reply:
yes, do you think it's more efficient (in Julia) ?

Reply:
I was more thinking for clarity. I'm sure any performance differences are negligible.